src/System.Linq/src/System/Linq/Enumerable.cs

```diff
@@ -2770,6 +2862,18 @@ public static decimal Average<TSource>(this IEnumerable<TSource> source, Func<TS
         }
     }
 
+    /// <summary>
+    /// An iterator that can (or sometimes can) produce an array through an optimised path.
+    /// </summary>
+    internal interface IArrayProvider<TElement>
+    {
+        /// <summary>
+        /// Produce an array of the sequence through an optimised path, if possible.
+        /// </summary>
+        /// <returns>The array, or null if an optimised path isn't possible, and default behaviour should be used.</returns>
```

Comment:
Sure, if I can tolerate Websterisms like "color" I can certainly accept where the US has the more traditional spelling :)
More seriously, I agree comments should be consistent as to dialect, that was just a slip.